internal/blueprint/fsnode_customizations.go

```diff
@@ -0,0 +1,284 @@
+package blueprint
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/osbuild/osbuild-composer/internal/common"
+	"github.com/osbuild/osbuild-composer/internal/fsnode"
+)
+
+// validateModeString checks that the given string is a valid mode octal number
+func validateModeString(mode string) error {
+	// check that the mode string matches the octal format regular expression
+	if regexp.MustCompile(`^0[0-7]{3}$`).MatchString(mode) {
+		return nil
+	}
```

Comment:
Are we sure we want to restrict it to having a leading `0`?  I suspect a lot of users will get tripped up by this from the general habit of `chmod`ing with just the three digits for convenience.

Reply:
I think I could change the regex to make the leading `0` optional. I agree that people are probably more used to write permissions without the leading zero.